projects/redis/redis/docker/linux/Dockerfile

```diff
@@ -0,0 +1,41 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG BASE_IMAGE
+ARG BUILDER_IMAGE
+FROM $BUILDER_IMAGE as redis-builder
+
+WORKDIR /
+RUN set -x && \
+    amazon-linux-extras enable redis6 && \
+    cp /etc/yum.repos.d/amzn2-extras.repo /newroot/etc/yum.repos.d/amzn2-extras.repo && \
+    clean_install systemd true true && \
+    clean_install redis && \
+    remove_package systemd true && \
```

Comment:
I wonder if we can clean this up some?  Typically in cases like this in the base images, we end up removing a number of packages that are not needed during runtime, such as bash/gawk/sed/other utils/etc.  I wonder if we know redis doesn't need those at run time?  I usually ldd on the binary to check for libraries it is loaded, but *mostly* you can remove the same packages we remove for nginx, something like [this](https://github.com/aws/eks-distro-build-tooling/blob/main/eks-distro-base/Dockerfile.minimal-base-nginx#L37) + bash (most likely)

For reference here are the packages currently included from the build log

```
#7 40.80 + cat /tmp/packages/redis
#7 40.80 audit-libs-2.8.1-3.amzn2.1.x86_64
#7 40.80 basesystem-10.0-7.amzn2.0.1.noarch
#7 40.80 bash-4.2.46-34.amzn2.x86_64
#7 40.80 bzip2-libs-1.0.6-13.amzn2.0.3.x86_64
#7 40.80 ca-certificates-2021.2.50-72.amzn2.0.1.noarch
#7 40.80 coreutils-8.22-24.amzn2.x86_64
#7 40.80 cracklib-2.9.0-11.amzn2.0.2.x86_64
#7 40.80 cracklib-dicts-2.9.0-11.amzn2.0.2.x86_64
#7 40.80 elfutils-default-yama-scope-0.176-2.amzn2.noarch
#7 40.80 elfutils-libelf-0.176-2.amzn2.x86_64
#7 40.80 elfutils-libs-0.176-2.amzn2.x86_64
#7 40.80 filesystem-3.2-25.amzn2.0.4.x86_64
#7 40.80 gawk-4.0.2-4.amzn2.1.2.x86_64
#7 40.80 glibc-2.26-56.amzn2.x86_64
#7 40.80 glibc-common-2.26-56.amzn2.x86_64
#7 40.80 glibc-minimal-langpack-2.26-56.amzn2.x86_64
#7 40.80 gmp-6.0.0-15.amzn2.0.2.x86_64
#7 40.80 gpg-pubkey-c87f5b1a-593863f8
#7 40.80 grep-2.20-3.amzn2.0.2.x86_64
#7 40.80 gzip-1.5-10.amzn2.x86_64
#7 40.80 info-5.1-5.amzn2.x86_64
#7 40.80 keyutils-libs-1.5.8-3.amzn2.0.2.x86_64
#7 40.80 krb5-libs-1.15.1-37.amzn2.2.2.x86_64
#7 40.80 libacl-2.2.51-14.amzn2.x86_64
#7 40.80 libattr-2.4.46-12.amzn2.0.2.x86_64
#7 40.80 libcap-2.22-9.amzn2.0.2.x86_64
#7 40.80 libcap-ng-0.7.5-4.amzn2.0.4.x86_64
#7 40.80 libcom_err-1.42.9-19.amzn2.x86_64
#7 40.80 libcrypt-2.26-56.amzn2.x86_64
#7 40.80 libdb-5.3.21-24.amzn2.0.3.x86_64
#7 40.80 libgcc-7.3.1-13.amzn2.x86_64
#7 40.80 libgcrypt-1.5.3-14.amzn2.0.2.x86_64
#7 40.80 libgpg-error-1.12-3.amzn2.0.3.x86_64
#7 40.80 libpwquality-1.2.3-5.amzn2.x86_64
#7 40.80 libselinux-2.5-12.amzn2.0.2.x86_64
#7 40.80 libsemanage-2.5-11.amzn2.x86_64
#7 40.80 libsepol-2.5-8.1.amzn2.0.2.x86_64
#7 40.80 libstdc++-7.3.1-13.amzn2.x86_64
#7 40.80 libverto-0.2.5-4.amzn2.0.2.x86_64
#7 40.80 logrotate-3.8.6-15.amzn2.x86_64
#7 40.80 lz4-1.7.5-2.amzn2.0.1.x86_64
#7 40.80 ncurses-6.0-8.20170212.amzn2.1.3.x86_64
#7 40.80 ncurses-base-6.0-8.20170212.amzn2.1.3.noarch
#7 40.80 ncurses-libs-6.0-8.20170212.amzn2.1.3.x86_64
#7 40.80 openssl-libs-1.0.2k-19.amzn2.0.10.x86_64
#7 40.80 pam-1.1.8-23.amzn2.0.1.x86_64
#7 40.80 pcre-8.32-17.amzn2.0.2.x86_64
#7 40.80 popt-1.13-16.amzn2.0.2.x86_64
#7 40.80 redis-6.2.6-1.amzn2.x86_64
#7 40.80 sed-4.2.2-5.amzn2.0.2.x86_64
#7 40.80 setup-2.8.71-10.amzn2.0.1.noarch
#7 40.80 shadow-utils-4.1.5.1-24.amzn2.0.2.x86_64
#7 40.80 system-release-2-14.amzn2.x86_64
#7 40.80 systemd-libs-219-78.amzn2.0.15.x86_64
#7 40.80 tzdata-2021a-1.amzn2.noarch
#7 40.80 ustr-1.0.4-16.amzn2.0.3.x86_64
#7 40.80 xz-libs-5.2.2-1.amzn2.0.2.x86_64
#7 40.80 zlib-1.2.7-18.amzn2.x86_64
```

bash + sed + shadow-utils + ncurses + gawk + grep stick out as possibly not needed.  What I have found before is most of these deps get included either from bash, or because they are needed during package install time, like shadow-utils, but not at runtime.